ftplugin/tex.lua

```diff
@@ -1,4 +1,42 @@
 require("lspconfig").texlab.setup {
   cmd = { DATA_PATH .. "/lspinstall/latex/texlab" },
   on_attach = require("lsp").common_on_attach,
+  handlers = {
+    ["textDocument/publishDiagnostics"] = vim.lsp.with(
+      vim.lsp.diagnostic.on_publish_diagnostics, {
+        virtual_text = O.lang.latex.diagnostics.virtual_text,
+        signs = O.lang.latex.diagnostics.signs,
+        underline = O.lang.latex.diagnostics.underline,
+        update_in_insert = true
+      })
+  },
+  -- filetypes = O.lang.latex.filetypes,
+  settings = {
+    texlab = {
+      auxDirectory = O.lang.latex.aux_directory,
+      bibtexFormatter = O.lang.latex.bibtex_formatter,
+      build = {
+        args = { '-pdf', '-interaction=nonstopmode', '-synctex=1', '%f' },
+        -- args = O.lang.latex.build.args,
```

Comment:
Either we make it configurable or not. It is misleading to have a O.lang.latex.build.args while changing it doesn't affect the configuration. I am in favor of keeping the configuration entry (for both compiler and viewer).

Reply:
I would love to have it configurable. But some how the arrays of the table in default-config.lua and lv-config.lua are not recognized. I'm new to lua maybe I'm making something wrong. The arrays which are affected are filetypes, texlab.build.args and texlab.forward_search.args.

For texlab.forward_search.args I try to implement a workaround. My plan is to write if statements which depend on the pdf-viewer, like zathura and so on.